src/v/cloud_storage/remote.cc

```diff
@@ -47,6 +48,41 @@ enum class error_outcome {
     notfound
 };
 
+/**
+ * Identify error cases that should be quickly retried, e.g.
+ * TCP disconnects, timeouts. Network errors may also show up
+ * indirectly as errors from the TLS layer.
+ */
+bool system_error_retryable(const std::system_error& e) {
+    auto v = e.code().value();
+
+    // The name() of seastar's gnutls_error_category class
+    constexpr std::string_view gnutls_cateogry_name{"GnuTLS"};
+
+    if (e.code().category().name() == gnutls_cateogry_name) {
```

Comment:
i wonder why seastar didn't put this in a header so we could check the category without string comparison :cry: